File: 13.js
// JavaScript
function minMax(nums) {
    // Handle empty array
    if (nums.length === 0) {
        return { min: null, max: null };
    }
    
    let min = nums[0];
    let max = nums[0];
    
    for (let i = 1; i < nums.length; i++) {
        if (nums[i] < min) {
            min = nums[i];
        }
        if (nums[i] > max) {
            max = nums[i];
        }
    }
    
    return { min, max };
}

// Examples
console.log(minMax([1, -3, 5]));  // { min: -3, max: 5 }
console.log(minMax([-2, 3, -5, 7, 10]));  // { min: -5, max: 10 }
console.log(minMax([]));  // { min: null, max: null }
console.log(minMax([3]));  // { min: 3, max: 3 }

File: 13.py
# Python
def min_max(nums):
    # Handle empty array
    if not nums:
        return {"min": None, "max": None}
    
    min_num = nums[0]
    max_num = nums[0]
    
    for num in nums[1:]:
        if num < min_num:
            min_num = num
        if num > max_num:
            max_num = num
    
    return {"min": min_num, "max": max_num}

# Examples
print(min_max([1, -3, 5]))  # {'min': -3, 'max': 5}
print(min_max([-2, 3, -5, 7, 10]))  # {'min': -5, 'max': 10}
print(min_max([]))  # {'min': None, 'max': None}
print(min_max([3]))  # {'min': 3, 'max': 3}